_pages/documentation/general_docs/kconfig_build_system.md

```diff
@@ -0,0 +1,206 @@
+
+```bash
+scons defconfig gem5_build build_opts/ALL
+```
+
+This will create a configuration for the `gem5_build` build directory based on
+`build_opts/ALL`. The configuration is stored in the
+`gem5_build/gem5.build/config`.
+
+To build a target in a configured build directory, run a scons command similar
+to before.
+
+```bash
+scons -j$(nproc) gem5_build/gem5.opt
+```
+
+Note: In order to have backward compatibility for old build scheme. The users
+need to avoid use "build" directory for Kconfig build.
+
+To build gem5 Kconfig with customized Kconfig options, the additional step
+is required between **initial configuration** and **building the target**
+
+- Directly set Kconfig options
+
+```bash
+scons setconfig gem5_build USE_KVM=y
+```
+
+- set Kconfig options by `menuconfig`
+
+```bash
+scons menuconfig gem5_build
+```
+
+## Details of Kconfig Commands
+
+### defconfig
+
+Set up a config using values specified in a defconfig file, or if no value is
+given, use the default. The second argument specifies the defconfig file. All
+of default gem5 defconfig files are in `build_opts` directory. The user can
+also use their own defconfig files.
+
+```bash
+scons defconfig gem5_build build_opts/RISCV
+```
+
+Use the own defconfig file
+
+```bash
+scons defconfig gem5_build $HOME/foo/bar/myconfig
+```
+
+### setconfig
+
+Set values in an existing config directory as specified on the command line.
+
+The users or developers can get the Kconfig options via `menuconfig` or `guiconfig`
+
+For example, to enable gem5's built in systemc kernel:
+
+```bash
+scons setconfig gem5_build USE_SYSTEMC=y
+```
+
+### menuconfig
+
+Opens the menuconfig editor which will let you view and edit config values,
+and view help text. menuconfig runs in CLI.
+
+```bash
+scons menuconfig gem5_build
+```
+
+If is successful, the CLI will look like this
+
+![](/assets/img/kconfig/menuconfig.png)
+
+To view help text, for example, USE_ARM_ISA, will look like this
+
+![](/assets/img/kconfig/menuconfig_details.png)
+
+If the `gem5_build` directory is not exists. SCons will set up a build
+directory at the path `gem5_build` with default options, and then invoke
+menuconfig so you can set up its configuration.
+
+### guiconfig
+
+Opens the guiconfig editor which will let you view and edit config values,
+and view help text. guiconfig runs as a graphical application.
+
+```bash
+scons guiconfig gem5_build
+```
+
+If is successful, it will create new windows to show up like
+
+![](/assets/img/kconfig/guiconfig.png)
+
+Open guiconfig in Ubuntu 22.04 with python3-tk
+
+### savedefconfig
+
+This helper utility lets you save the defconfig which would give rise to
+a given config. For instance, you could use menuconfig to set up a
+config how you want it with the options you cared about, and then use
+savedefconfig to save a minimal config file. These files would be suitable to
+use in the build_opts directory. The second argument specifies the filename
+for the new defconfig file.
```

Comment:
suggested grammar change: 
This helper utility allows you to save the defconfig that corresponds to a given configuration. For instance, you can use menuconfig to set up a configuration with the options you care about, and then use savedefconfig to create a minimal configuration file. These files are suitable for use in the `build_opts` directory. The second argument specifies the filename for the new defconfig file.